uwdata/lyra
# Walkthrough setup

Walkthroughs are here and boy are they... ... well they aren't the most attractive thing but they have :sparkles: potential :sparkles: 

This PR includes the following:
- A new fancy footer and modal!  You can probably close PR #339 if you merge this in.
- A walkthrough menu for selecting walkthroughs
- some redux reducers for storing the walkthrough state
- a new hints box that needs some css love but hey, it's ok for now
- the ugliest error messaging you've ever seen
- a lot of Bill Murray.  There are 2 example walkthroughs, that exist in /walkthrough/files.   Am I in love with this folder structure? nope.
- I am not loading the walkthrough json through "thunks", I am just loading them through require since they exist inside the app. I thought about this and I don't see any reason you might host them outside of the app.

![walkthrough](https://cloud.githubusercontent.com/assets/607541/14576603/60624f3e-031f-11e6-8d06-22139d4f2cf7.gif)

**This PR includes:**

**Steps to functionally test this:**
- click on the WALKTHROUGHS link in the footer
- select a walkthrough
- follow the prompted steps, or don't follow them and get errors. It's your life and I'm not gonna tell you how to live it.  

The validator is very forgiving. It's only going to check that the things you asked for in the walkthrough spec exist, but if you do extra fancy things like change colors, or fonts or add 17 line marks, it's not going to yell at you as long as you meet the requirement of the prompts. That's by design.
